pkg/operator/credentialsrequest/credentialsrequest_controller.go

```diff
@@ -300,13 +306,54 @@ func (r *ReconcileCredentialsRequest) Reconcile(request reconcile.Request) (reco
 		logger.WithError(err).Error("error getting credentials request, requeuing")
 		return reconcile.Result{}, err
 	}
-	logger = logger.WithFields(log.Fields{
-		"secret": fmt.Sprintf("%s/%s", cr.Spec.SecretRef.Namespace, cr.Spec.SecretRef.Name),
-	})
+
 	// Maintain a copy, but work on a copy of the credentials request:
 	origCR := cr
 	cr = cr.DeepCopy()
 
+	// Copy deprecated secretRef to the new storage field for internal processing
+	if cr.Spec.StorageDestination.Type == "" {
+		cr.Spec.StorageDestination.Type = minterv1.SecretStorageDestinationType
+	}
+	cr.Spec.StorageDestination.Secret = &minterv1.SecretStorageDestination{
+		Namespace: cr.Spec.SecretRef.Namespace,
+		Name:      cr.Spec.SecretRef.Name,
+	}
+	storageType := cr.Spec.StorageDestination.Type
+
+	if storageType != minterv1.SecretStorageDestinationType && storageType != minterv1.ServiceAccountStorageDestinationType {
+		setInvalidStorageType(cr, r.platformType)
+		err := r.updateStatus(origCR, cr, logger)
+		if err != nil {
+			logger.WithError(err).Error("failed to update conditions")
+		}
+		return reconcile.Result{}, err
+	}
+
+	// CRs must contain a reference for their storage type (union)
+	if (storageType == minterv1.SecretStorageDestinationType && cr.Spec.StorageDestination.Secret == nil) ||
```

Comment:
how would spec.storageDestination.secret ever be nil if we unconditionally populate it a few lines above?

Reply:
yes, i moved the setting of the secret inside the `if type == ""` block above